addons/TreasurePool/TreasurePool.lua

```diff
@@ -0,0 +1,171 @@
+        if packet.Item == 0xFFFF then
+            return
+        end
+        if diff <= 300 then
+            goals[packet.Index] = packet.Timestamp + 300
+            lotter[packet.Index] = ' '
+        else
+            goals[packet.Index] = os.time() + 300
+            lotter[packet.Index] = ' '
+        end
+    
+    end
+    
+    if id == 0x0D3 then
+ 
+    local lotpacket = packets.parse('incoming', data)
+ 
+        -- Ignore drop to a player or floored
+        if lotpacket.Drop ~= 0 then
+            return
+        else    
+            lotter[lotpacket.Index] = lotpacket['Highest Lotter Name']
+            lot[lotpacket.Index] = lotpacket['Highest Lot']
+        end
+    
+    end
+    
+    -- Check to hide text box if zoning with treasure up
+    if id == 0xB then
+        zoning_bool = true
+    elseif id == 0xA and zoning_bool then
+        zoning_bool = nil
+    end
+    
+end)
+
+windower.register_event('prerender', function()
+    local get_item = windower.ffxi.get_items()
+    local remove = S{}
+    local info = {}
+    if zoning_bool or (not get_item.treasure[0] and not get_item.treasure[1] and not get_item.treasure[2] and not get_item.treasure[3] and not get_item.treasure[4] and not get_item.treasure[5] and not get_item.treasure[6] and not get_item.treasure[7] and not get_item.treasure[8] and not get_item.treasure[9]) then
+        remove:add('index0') remove:add('index1') remove:add('index2') remove:add('index3') remove:add('index4') remove:add('index5') remove:add('index6') remove:add('index7') remove:add('index8') remove:add('index9')
+        remove:add('lotting0') remove:add('lotting1') remove:add('lotting2') remove:add('lotting3') remove:add('lotting4') remove:add('lotting5') remove:add('lotting6') remove:add('lotting7') remove:add('lotting8') remove:add('lotting9')
+        treasure:update(info)
+        treasure:hide()
+        return
+    end
+    for i,v in pairs(get_item.treasure) do
+        for i = 0, 9 do
+            if get_item.treasure[i] then
+                if goals[i] then
+                    local diff = os.difftime(goals[i], os.time())
+                    local timer = {}
+                    local item = {}
+                    item[i] = res.items[get_item.treasure[i].item_id].name
+                    timer[i] = os.date('!%M:%S', diff)
+                    if diff < 0 then -- stop the timer when 00:00 so it don't show 59:59 for a brief moment
+                        remove:add('index' .. i)
+                        remove:add('lotting' .. i)
+                    else
+                        info['index' .. i] = (
+                            diff < 60 and
+                                '\\cs(255,0,0)' .. item[i] .. ' → ' .. timer[i]
+                            or diff > 180 and
+                                '\\cs(0,255,0)' .. item[i] .. ' → ' .. timer[i]
+                            or
+                                '\\cs(255,128,0)' .. item[i] .. ' → ' .. timer[i]) .. '\\cr'
```

Comment:
This can be done prettier, but I'll need to explain a few other things first. Hit me up on IRC about it at some point during the weekend.